### PR TITLE
Rename namespace and Timex class

### PR DESCRIPTION
**Pull request by johnataylor**
namespace becomes MIcrosoft.Recognizers.Text.DataTypes.TimexExpression
change is reflected everywhere... package, project, folder etc.

class Timex becomes class TimexProperties which is probably a more accurate name.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/Timex.cs [hunk 1]
```diff
@@ -3,32 +3,32 @@
 
 using System.Collections.Generic;
 
-namespace Microsoft.Recognizers.DataTypes.DateTime
+namespace Microsoft.Recognizers.Text.DataTypes.TimexExpression
 {
-    public class Timex
+    public class TimexProperties
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/Timex.cs]: I always prefer that the file name matches the class name.

File: .NET/CreateAllPackages.cmd [hunk 2]
```diff
@@ -30,7 +30,7 @@ popd
 pushd Microsoft.Recognizers.Text.Choice
 call CreatePackage.cmd
 popd
-pushd Microsoft.Recognizers.DataTypes.DateTime
+pushd Microsoft.Recognizers.Text.DataTypes.TimexExpression
```
Comment (tellarin) [inline .NET/CreateAllPackages.cmd]: The new namespace could be Microsoft.Recognizers.Text.DataTypes.TimeExpression, without the X in Timex as it's redundant. But I'm fine with keeping it.

File: .NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/Timex.cs (hunk 1 above)
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DataTypes.TimexExpression/Timex.cs]: Shouldn't TimexProperties be singluar?

File: .NET/CreateAllPackages.cmd (hunk 2 above)
Comment (johnataylor) [reply .NET/CreateAllPackages.cmd]: should be good now

**Pull request merged**